ui/components/src/StatChart/StatChart.tsx

```diff
@@ -40,14 +43,43 @@ export interface StatChartProps {
   unit: UnitOptions;
   color?: string;
   sparkline?: LineSeriesOption;
+  showSeriesName?: boolean;
 }
 
 export function StatChart(props: StatChartProps) {
-  const { width, height, data, unit, color, sparkline } = props;
+  const { width, height, data, unit, color, sparkline, showSeriesName } = props;
   const chartsTheme = useChartsTheme();
-
   const formattedValue = data.calculatedValue === undefined ? '' : formatValue(data.calculatedValue, unit);
 
+  const containerPadding = chartsTheme.container.padding.default;
+
+  // calculate series name font size and height
+  let seriesNameFontSize = useOptimalFontSize({
+    text: data?.seriesData?.name ?? '',
+    fontWeight: SERIES_NAME_FONT_WEIGHT,
+    width,
+    height: height * 0.125, // assume series name will take 12.5% of available height
```

Comment:
Out of curiosity, where did 12.5% come from? Is it an approximation or is it based on some specific math?

Reply:
from @foxbat07, all these percentages are just approximation for how we want the stat chart to look